crates/revm/src/db/states/bundle_state.rs

```diff
@@ -658,6 +658,23 @@ impl BundleState {
             }
         }
     }
+
+
+    /// Prepends present the state with the given BundleState.
+    /// It adds changes from the given state but does not override any existing changes.
+    ///
+    /// Reverts are not updated.
+    pub fn prepend_state(&mut self, mut other: BundleState) {
+        let other_len = other.reverts.len();
+        // take this bundle
+        let this_bundle = std::mem::take(self);
+        // extend other bundle with this
+        other.extend(this_bundle);
+        // discard other reverts
+        other.take_n_reverts(other_len);
+        // swap bundles
+        std::mem::swap(self, &mut other)
```

Comment:
This works, but we could be better with a small refactor. Make a `fn extend_state` and make `fn extend` calls it (it is just moving some code), and with this, you would not need to `take_n_reverts` and could extend only needed data.